Meta/check-png-sizes.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+# How many bytes optipng has to be able to strip out of the file for the optimization to be worth it. The default is 2 KiB.
+MINIMUM_OPTIMIZATION_BYTES="${MINIMUM_OPTIMIZATION_BYTES:-1024}"
+
+script_path=$(cd -P -- "$(dirname -- "$0")" && pwd -P)
+cd "${script_path}/.."
+
+if ! command -v optipng >/dev/null ; then
+    echo 'optipng is not installed, skipping png size check.'
+    echo 'Please install optipng for your system to run this check.'
+    exit 0
+fi
+
+files=()
+for file in "$@"; do
+    if [[ "${file}" == *".png" ]]; then
+        files+=("${file}")
+    fi
+done
+
+if (( ${#files[@]} )); then
+    # We need to allow optipng to write output so we can check what it actually did. We use a dummy file that's discarded afterwards.
+    optimizations=$( printf '%s\0' "${files[@]}" |\
+        xargs -0 -n1 optipng -out dummy-optipng-output.png -clobber 2>&1 |\
+        grep -i -e 'Output IDAT size =' |\
+        sed -E 's/Output IDAT size = [0-9]+ byte(s?) \(([0-9]+) byte(s?) decrease\)/\2/g;s/Output IDAT size = [0-9]+ byte(s?) \(no change\)/0/g' |\
+        awk "{ if (\$1 >= $MINIMUM_OPTIMIZATION_BYTES) { S+=\$1 } } END { print S }")
+    rm dummy-optipng-output.png || true
+    rm dummy-optipng-output.png.bak || true
+    optimizations="${optimizations:-0}"
+
+    if [ "$optimizations" != 0 ] ; then
+        echo "There are non-optimized PNG images in Base/. It is possible to reduce file sizes by at least $optimizations byte(s)."
+        echo 'Please run optipng with default configuration on modified PNG images and try again.'
```

Comment:
I've found the defaults to still lead to much bigger images than `-strip all` - with the downside that `-strip all` sometimes produces images our PNG decoder chokes on (usually related to palette stuff, perhaps solvable with `-np`)...

Reply:
I'd say let's leave it for now, especially if that's the case.

Reply:
That seems like a worse outcome compared what we have now (mostly me telling people to run with `-strip all`, which works out fine in 9/10 cases, and using a less aggressive option in every 10th case)

Reply:
okay fine I guess